### PR TITLE
add support for the CentOS and Fedora repo

### PR DESCRIPTION
**Pull request by nathenharvey**
I've added support for the CentOS and Fedora repo.

Let me know if you have any questions.

### CODE REVIEW

**Comment by thekorn** [comment]
Wow Nathen, another feature which is nice to have, thanks for that.
Unfortunatly I cannot tests it, because I don't use these systems, so I have one question: is it working? There is esp. one thing I'm concerned about: the cookbook ships an init (service) script (created in /etc/init.d/mongo*), and configures the services using files in /etc/default. Is this working in centos etc? Or do we need to define services in a different way there? 

Markus

**Comment by nathenharvey** [comment]
I will try to do some additional testing over the next week or so to 
verify it works.  I only added support for the CentOS repo, didn't 
actually test out using the cookbook you have on CentOS.

**Comment by looztra** [comment]
@nathenharvey : you may need to make your recipe install the package **mongodb-10gen-server** too because the package  **mongodb-10gen** only installs the mongodb client. More over, @thekorn's default recipe also needs some tuning because there is a call to a package install with the name **mongodb** that won't work under redhat/centos.

@thekorn : I 'm currently experimenting you recipe under centos, also preparing a pull request. stay tuned :)

File: mongodb/recipes/10gen_repo.rb [hunk 1]
```diff
@@ -42,6 +42,27 @@
   package "mongodb" do
     package_name "mongodb-10gen"
   end
+
+when "centos","redhat","fedora","suse"
```
Comment (dcrosta) [inline mongodb/recipes/10gen_repo.rb]: the RPM works on "amazon" platforms as well

File: mongodb/recipes/10gen_repo.rb [hunk 2]
```diff
@@ -42,6 +42,27 @@
...
+	  group "root"
+	  mode "0644"
+	  variables(
+	    :processor => processor
+	  )
+	end
+
+	execute "refresh repos" do
+	  command "yum clean all"
+	end
```
Comment (dcrosta) [inline mongodb/recipes/10gen_repo.rb]: All of the above could be replaced with

```
yum_repository "10gen" do
  description "10gen RPM Repository"
  url "http://downloads-distro.mongodb.org/repo/redhat/os/$basearch"
  action :add
end
```

The `$basearch` isn't Chef magic; that's literally in the file, and yum resolves it when it reads the URL.

Also the cookbook will have to depend on the "yum" cookbook which has this LWRP.

File: mongodb/README.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ Installs and configures MongoDB, supporting:
 
 The cookbook aims to be platform independant, but is best tested on debian squeeze systems.
 
-The `10gen_repo` recipe currently supports only the Debian and Ubuntu 10gen repository.
+The `10gen_repo` recipe currently supports only the Debian and Ubuntu and Fedora and CentOS 10gen repository.
```
Comment (miketheman) [inline mongodb/README.md:17]: Maybe remove 'only' - since this would now support both repositories.

**Comment by miketheman** [comment]
@nathenharvey - Any chance you can take a shot at Dan's comments, and possibly reissue the pull against the updated repo? This cannot be automatically merged as-is.

File: mongodb/recipes/10gen_repo.rb (hunk 2 above)
Comment (miketheman) [inline mongodb/recipes/10gen_repo.rb]: The yum variable `$arch` should probably be used instead of `$basearch`, since the repo uses i686 instead of i386.

Comment (dcrosta) [inline mongodb/recipes/10gen_repo.rb]: Thanks -- updated https://github.com/customink/chef-cookbooks/pull/1 which will show up here when CustomInk merges on their end.

**Comment by dcrosta** [comment]
Final request for comments from others -- if no one speaks up, I'll merge this EOD today.

cc @miketheman @thekorn @tobami @patcon @jamesonjlee

**Comment by dcrosta** [comment]
Thanks @nathenharvey!

**Pull request merged**